observability/src/main/java/io/grpc/observability/interceptors/ConfigFilterHelper.java

```diff
@@ -0,0 +1,217 @@
+        FilterParams params = FilterParams.create(true,
+            currentHeaderBytes, currentMessageBytes);
+        perMethodFilters.put(methodOrServicePattern, params);
+      }
+    }
+    this.globalLog = globalLog;
+    this.globalParams = globalParams;
+    this.perServiceFilters = ImmutableMap.copyOf(perServiceFilters);
+    this.perMethodFilters = ImmutableMap.copyOf(perMethodFilters);
+    this.methodOrServiceFilterPresent = true;
+  }
+
+  @VisibleForTesting
+  void setEventFilterSet() {
+    List<EventType> eventFilters = config.getEventTypes();
+    if (eventFilters == null) {
+      return;
+    }
+    this.logEventTypeSet = ImmutableSet.copyOf(eventFilters);
+  }
+
+  /**
+   * Class containing results for method/service filter information, such as flag for logging
+   * method/service and payload limits to be used for filtering.
+   */
+  @AutoValue
+  public abstract static class FilterParams {
+    abstract boolean log();
+
+    abstract int headerBytes();
+
+    abstract int messageBytes();
+
+    @VisibleForTesting
+    public static FilterParams create(boolean log, int headerBytes, int messageBytes) {
+      return new AutoValue_ConfigFilterHelper_FilterParams(
+          log, headerBytes, messageBytes);
+    }
+  }
+
+  /**
+   * Checks if the corresponding service/method passed needs to be logged as per the user provided
+   * configuration.
+   *
+   * @param method the fully qualified name of the method
+   * @return MethodFilterParams object 1. specifies if the corresponding method needs to be logged
+   *     (log field will be set to true) 2. values of payload limits retrieved from configuration
+   */
+  public FilterParams isMethodToBeLogged(MethodDescriptor<?, ?> method) {
+    FilterParams params = NO_FILTER_PARAMS;
+    String fullMethodName = method.getFullMethodName();
+    if (methodOrServiceFilterPresent) {
+      if (!perMethodFilters.isEmpty() && perMethodFilters.containsKey(fullMethodName)) {
+        params = perMethodFilters.get(fullMethodName);
+      } else {
+        String serviceName = method.getServiceName();
+        if (!perServiceFilters.isEmpty() && perServiceFilters.containsKey(serviceName)) {
+          params = perServiceFilters.get(serviceName);
+        } else if (globalLog) {
+          params = globalParams;
+        }
+      }
+    }
+    return params;
+  }
+
+  /**
+   * Checks if the corresponding event passed needs to be logged as per the user provided
+   * configuration.
+   *
+   * <p> All events are logged by default if event_types is not specified or {} in configuration.
+   * If event_types is specified, only the events specified in the configuration will be logged.
+   * </p>
+   *
+   * @param event gRPC observability event
+   * @return true if event needs to be logged, false otherwise
+   */
+  public boolean isEventToBeLogged(EventType event) {
+    boolean logEvent = true;
+    if (!logEventTypeSet.isEmpty()) {
+      logEvent = logEventTypeSet.contains(event);
```

Comment:
Just realized there are 3 possibilities:

- the config has no `event_types` field e.g. config is just `{}`. In this case `logEventTypeSet` ought to be `null`
- the config has `event_types` field but is empty set. e.g. `{ "event_types": []}`. In this case `logEventTypeSet` ought to be empty.
- the config has `event_types` field and is not empty e.g. `{ "event_types": ["GRPC_CALL_HALF_CLOSE"]}`. In this case `logEventTypeSet` is not empty.


The code here (and before) does not differentiate between case 1 and 2.

Reply:
Made the changes as below:
- log all events when `event_types` is `{}`
- do not log any events when `event_types` is `[]`, assuming user does not want any events to be logged and explicitly set it to be empty
- log events specified in non empty `event_types`